controller/service.go

```diff
@@ -103,11 +103,20 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 		// The user might also have changed the pool annotation, and
 		// requested a different pool than the one that is currently
 		// allocated.
-		desiredPool := svc.Annotations[annotationAddressPool]
-		if len(lbIPs) != 0 && desiredPool != "" && c.ips.Pool(key) != desiredPool {
-			level.Info(l).Log("event", "clearAssignment", "reason", "differentPoolRequested", "msg", "user requested a different pool than the one currently assigned")
-			c.clearServiceState(key, svc)
-			lbIPs = []net.IP{}
+		desiredPoolStr := svc.Annotations[annotationAddressPool]
+		if len(lbIPs) != 0 && desiredPoolStr != "" {
```

Comment:
Let's split this in two too. One check is, the user requested a different pool, so we clean.
The other is, we check if the pool is compatible and if not we exit